utils.js

```diff
@@ -1,263 +1,295 @@
+  else if (fetchResponse && ((fetchResponse.headers.link) || lowerCase(url) !== lowerCase(fetchResponse.url) || (fetchResponse.headers['www-authenticate']))) {
+    fetch = 'bad fetch';
+  }
 
-  return toConvert.then(files => {
-    let fileCount = 0;
+  return fetch;
+};
 
-    //Loop through the file contents and add them to the new zip archive
-    //with the file names and buffers associated with them.
-    files.forEach(file => {
-      //Ignore folders in the total file count, as we
-      //do not care about indexing folders.
-      if (file.type !== 'directory') {
-        fileCount++;
-      }
+//Function to check if the total number of files
+//processed exceeds 50.
+const fileCountChecker = fileCount => {
+  if(fileCount > 50){
+    throw new Error(messages.TOO_MANY_FILES);
+  }
+  return false;
+};
 
-      //Too many files in archive, catch early and throw an error
-      if (fileCount > 50) {
-        throw new Error(messages.TOO_MANY_FILES);
-      }
 
-      newArchive.add(file.path, file.data);
-    });
-    //Turn the entire zip in a zip file buffer, and send off to the
-    //zip file handler.
-    archiveFile.content = newArchive.toBuffer();
-    return handleZip(archiveFile);
-  });
+//Function to check if the file size or total size of the files
+//fetched exceeds 100 MB. If no size can be fetched, find it from
+//the buffer first, then check.
+const fileSizeChecker = (fileSize, fileBuffer) => {
+  if(fileSize === 'null' && fileBuffer){
+    fileSize = getStringByteSize(fileBuffer);
+  } 
+  if(fileSize > 100 * 1024 * 1024){
+    throw new Error(messages.BIG_FILE);
+  }
+  return fileSize;
 };
 
-//This is a function to determine the type of compression used on a file. If none of these are detected, it will be indexed with whichever default it gets.
-// If the data is in a zip but not compressed, give it the uncompressed data compression, otherwise it is deflate for zip. Check the first few bits to see if other algorithms used
-//otherwise default to either uncompressed or deflate. If no uncompressedSize is provided in the parameters, a zip file content is being looked at, so default to deflate, otherwise default uncompressed.
-// Used this as a reference: https://stackoverflow.com/questions/19120676/how-to-detect-type-of-compression-used-on-the-file-if-no-file-extension-is-spe for compression detecting
-const findCompressionType = (zipContent, uncompressedSize) => {
 
-  //Default
-  let compressionType = 'UNCOMPRESSED';
+//Function for finding the filetype/extension of a file after it's contents have been fetched
+const findExtension = (file, response) => {
+  const mime = require('mime-types');
+  const path = require('path');
+  let fileExtension = '';
+
+  //If the filename is trying to be determined from node-fetch call, aka when
+  //a file is being looked at that isn't inside of an archive file
+  if(response){
+    fileExtension = '.' + mime.extension(response.headers.get('content-type'));
```

Comment:
It seems to have many detours here with the extension because you cast it in a string 
If mime.extension() is return false when it fails, you should take advantage of that.

Something like
```javascript
    fileExtension = mime.extension(response.headers.get('content-type'));
     
    //If mime-type couldn't pick up an extension, try file-type to get it
    //based on the data buffer
    if ( !fileExtension ) {
      if(fileType(fileExtension) === null){ //Bad call, do nothing
      } else {
        fileExtension = fileType(file.content).ext;
      }
    } 
  
    //If mime-type and file-type fail, last resort is the file name for an extension.
    if( !fileExtension ){
      fileExtension = path.extname(file.filename);
    } 
  
    if ( fileExtension ) {
      fileExtension = '.' + fileExtension;
    }
     return fileExtension;

```